rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -141,14 +152,40 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
   }
 
   template<typename T>
-  typename std::enable_if<std::is_same<T, rcl_serialized_message_t>::value, void>::type
+  typename std::enable_if<
+    std::is_same<T, rcl_serialized_message_t>::value &&
+    !std::is_same<MessageT, rclcpp::experimental::SerializedContainer>::value,
+    void>::type
   execute_impl()
   {
-    throw std::runtime_error("Subscription intra-process can't handle serialized messages");
+    if (serializer_) {
+      rmw_message_info_t msg_info;
+      msg_info.from_intra_process = true;
+
+      ConstMessageSharedPtr msg = buffer_->consume_shared();
+      auto serialized_msg =
+        serializer_->serialize_message(reinterpret_cast<const void *>(msg.get()));
+
+      if (serialized_msg == nullptr) {
+        throw std::runtime_error("Subscription intra-process could not serialize message");
+      }
+
+      if (any_callback_.use_take_shared_method()) {
+        any_callback_.dispatch_intra_process(serialized_msg, msg_info);
+      } else {
+        throw std::runtime_error("Subscription intra-process for serialized "
+                "messages does not support unique pointers.");
+      }
+    } else {
+      throw std::runtime_error("Subscription intra-process can't handle serialized messages");
+    }
   }
 
   template<class T>
-  typename std::enable_if<!std::is_same<T, rcl_serialized_message_t>::value, void>::type
+  typename std::enable_if<
+    !std::is_same<T, rcl_serialized_message_t>::value &&
+    !std::is_same<MessageT, rclcpp::experimental::SerializedContainer>::value,
```

Comment:
Connecting to the comment above, this specialization is `publish(rcl_serialized_msg_t) -> subscribe(rcl_serailized_msg_t)` ?